pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -319,6 +319,8 @@ func (e *EndpointController) syncService(key string) error {
 
 	namespace, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Syncing service %q endpoints, but couldn't understand the key: %v", key, err))
```

Comment:
Can you change the error message to:

fmt.Errorf("syncService(%q), invalid key: %v", key, err)